### PR TITLE
Update policer rate for cisco-8000

### PR DESCRIPTION
**Pull request by zhixzhu**
### Description of PR

Summary:
Fixes # (issue)
MIGSMSFT-614 [81xx] test_copp.py::test_policer failed with LLDP

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [ ] 202405

#### How did you verify/test it?
8111 T1: 
SONiC Software Version: SONiC.azure_cisco_202311.14115-dirty-20240730.104536
```
--------------------------------------------------- generated xml file: /tmp/copp/test_copp.py::TestCOPP::test_policer_2024-08-30-21-27-41.xml ----------------------------------------------------
INFO:root:Can not get Allure report URL. Please check logs
------------------------------------------------------------------------------------- live log sessionfinish --------------------------------------------------------------------------------------
21:49:36 __init__.pytest_terminal_summary         L0067 INFO   | Can not get Allure report URL. Please check logs
===================================================================================== short test summary info =====================================================================================
PASSED copp/test_copp.py::TestCOPP::test_policer[croc-aaa12-dut-ARP]
PASSED copp/test_copp.py::TestCOPP::test_policer[croc-aaa12-dut-IP2ME]
PASSED copp/test_copp.py::TestCOPP::test_policer[croc-aaa12-dut-SNMP]
PASSED copp/test_copp.py::TestCOPP::test_policer[croc-aaa12-dut-SSH]
PASSED copp/test_copp.py::TestCOPP::test_policer[croc-aaa12-dut-DHCP]
PASSED copp/test_copp.py::TestCOPP::test_policer[croc-aaa12-dut-DHCP6]
PASSED copp/test_copp.py::TestCOPP::test_policer[croc-aaa12-dut-BGP]
PASSED copp/test_copp.py::TestCOPP::test_policer[croc-aaa12-dut-LACP]
PASSED copp/test_copp.py::TestCOPP::test_policer[croc-aaa12-dut-LLDP]
PASSED copp/test_copp.py::TestCOPP::test_policer[croc-aaa12-dut-UDLD]
=========================================================================== 10 passed, 1 warning in 1314.41s (0:21:54) ============================================================================
sonic@sonic-ucs-m5-8:/data/tests$ 
```

8102 T0:
SONiC.azure_cisco_202305.15147-dirty-20240828.020736
```
--------------------------------------------------- generated xml file: /tmp/copp/test_copp.py::TestCOPP::test_policer_2024-08-31-00-18-10.xml ----------------------------------------------------
INFO:root:Can not get Allure report URL. Please check logs
------------------------------------------------------------------------------------- live log sessionfinish --------------------------------------------------------------------------------------
00:34:34 __init__.pytest_terminal_summary         L0067 INFO   | Can not get Allure report URL. Please check logs
===================================================================================== short test summary info =====================================================================================
PASSED copp/test_copp.py::TestCOPP::test_policer[mth-t0-64-ARP]
PASSED copp/test_copp.py::TestCOPP::test_policer[mth-t0-64-IP2ME]
PASSED copp/test_copp.py::TestCOPP::test_policer[mth-t0-64-SNMP]
PASSED copp/test_copp.py::TestCOPP::test_policer[mth-t0-64-SSH]
PASSED copp/test_copp.py::TestCOPP::test_policer[mth-t0-64-DHCP]
PASSED copp/test_copp.py::TestCOPP::test_policer[mth-t0-64-DHCP6]
PASSED copp/test_copp.py::TestCOPP::test_policer[mth-t0-64-BGP]
PASSED copp/test_copp.py::TestCOPP::test_policer[mth-t0-64-LACP]
PASSED copp/test_copp.py::TestCOPP::test_policer[mth-t0-64-LLDP]
PASSED copp/test_copp.py::TestCOPP::test_policer[mth-t0-64-UDLD]
=========================================================================== 10 passed, 11 warnings in 982.55s (0:16:22) ===========================================================================
sonic@sonic-ucs-m6-2:/data/tests$ 
```

8101 T0:
SONiC.azure_cisco_202305.15147-dirty-20240828.020736
```
--------------------- generated xml file: /tmp/copp/test_copp.py::TestCOPP::test_policer_2024-08-31-05-46-04.xml ----------------------
INFO:root:Can not get Allure report URL. Please check logs
------------------------------------------------------- live log sessionfinish --------------------------------------------------------
06:00:35 __init__.pytest_terminal_summary         L0067 INFO   | Can not get Allure report URL. Please check logs
======================================================= short test summary info =======================================================
PASSED copp/test_copp.py::TestCOPP::test_policer[cmono-t0-dut-ARP]
PASSED copp/test_copp.py::TestCOPP::test_policer[cmono-t0-dut-IP2ME]
PASSED copp/test_copp.py::TestCOPP::test_policer[cmono-t0-dut-SNMP]
PASSED copp/test_copp.py::TestCOPP::test_policer[cmono-t0-dut-SSH]
PASSED copp/test_copp.py::TestCOPP::test_policer[cmono-t0-dut-DHCP]
PASSED copp/test_copp.py::TestCOPP::test_policer[cmono-t0-dut-DHCP6]
PASSED copp/test_copp.py::TestCOPP::test_policer[cmono-t0-dut-BGP]
PASSED copp/test_copp.py::TestCOPP::test_policer[cmono-t0-dut-LACP]
PASSED copp/test_copp.py::TestCOPP::test_policer[cmono-t0-dut-LLDP]
PASSED copp/test_copp.py::TestCOPP::test_policer[cmono-t0-dut-UDLD]
============================================== 10 passed, 1 warning in 870.15s (0:14:30) ==============================================
sonic@sonic-ucs-m6-13:/data/tests$ 
```

### CODE REVIEW

**Review by alpeshspatel — APPROVED** [review]

**Review by wsycqyz — APPROVED** [review]
lğtm

File: ansible/roles/test/files/ptftests/py3/copp_tests.py [hunk 1]
```diff
@@ -352,10 +353,18 @@ def __init__(self):
...
+        # Cisco G100 based platform has CIR 600
+        elif self.asic_type == "cisco-8000" and "8111" in self.platform:
+            self.PPS_LIMIT = 600
         else:
             self.PPS_LIMIT = 100
-        self.PPS_LIMIT_MIN = self.PPS_LIMIT * 0.9
-        self.PPS_LIMIT_MAX = self.PPS_LIMIT * 1.3
+        if self.asic_type == "cisco-8000" and any(platform in self.platform
+                                                  for platform in ["8102", "8101", "8111"]):
+            self.PPS_LIMIT_MIN = self.PPS_LIMIT * 0.8
```
Comment (kevinskwang) [inline ansible/roles/test/files/ptftests/py3/copp_tests.py:363]: why we need change here?

File: ansible/roles/test/files/ptftests/py3/copp_tests.py [hunk 2]
```diff
@@ -396,10 +405,18 @@ def __init__(self):
...
```
Comment (kevinskwang) [inline ansible/roles/test/files/ptftests/py3/copp_tests.py:415]: same as above

File: ansible/roles/test/files/ptftests/py3/copp_tests.py (hunk 1 above)
Comment (zhixzhu) [reply ansible/roles/test/files/ptftests/py3/copp_tests.py:363]: In the investigation on the meter profiles and calculation on the profile parameters. Will update later.

Comment (zhixzhu) [reply ansible/roles/test/files/ptftests/py3/copp_tests.py]: Removed this change. Instead I changed the CIR configuration.

In Q200 meter profiles, minimum CBS is 1024, and minimum EBS is 2048.
For 30 seconds' traffic, if CIR is 100, then the expected result is:
Green: 1024 + 100 * 30 = 4024
Yellow: 2048
Theoretical rate of punted packets: (4024 + 2048) / 30 = 202.4, which is outside of the range [0.9 * 100, 1.3 * 100].

To fall in the script pass criteria [0.9 * CIR, 1.3 * CIR], set the CIR 400 pps.
Green: 1024 + 400 * 30 = 13024
Yellow: 2048
Theoretical rate of punted packets: (13024 + 2048) / 30 = 502.4, which is < 400 * 1.3.

File: ansible/roles/test/files/ptftests/py3/copp_tests.py (hunk 2 above)
Comment (zhixzhu) [reply ansible/roles/test/files/ptftests/py3/copp_tests.py]: This change is removed.

**Comment by alpeshspatel** [comment]
@kevinskwang Can you please do the needful. All comments addressed and All checks have passed.

**Comment by kevinskwang** [comment]
@zhixzhu  pls address the conflict

**Comment by zhixzhu** [comment]
@kevinskwang conflict resolved.

**Comment by alpeshspatel** [comment]
@kevinskwang  can you please do the needful.

**Review by kevinskwang — APPROVED** [review]

File: tests/copp/copp_utils.py [hunk 3]
```diff
@@ -63,11 +63,12 @@ def limit_policer(dut, pps_limit, nn_target_namespace):
         config_format = "config_db"
 
     dut.script(
-        cmd="{} {} {} {} {}".format(_UPDATE_COPP_SCRIPT,
-                                    pps_limit,
-                                    _BASE_COPP_CONFIG,
-                                    _TEMP_COPP_CONFIG,
-                                    config_format)
+        cmd="{} {} {} {} {} {}".format(_UPDATE_COPP_SCRIPT,
```
[the remark this answers is not in the document]
Comment (zhixzhu) [reply tests/copp/copp_utils.py:66]: Added one parameter asic_type.

**Comment by zhixzhu** [comment]
> @zhixzhu PR conflicts with 202305 branch

PR to double commit to 202305 branch: https://github.com/sonic-net/sonic-mgmt/pull/15241

**Comment by zhixzhu** [comment]
@kevinskwang This change is also needed in 202311 branch.

**Comment by kevinskwang** [comment]
No need as it is 8111.

**Pull request merged**